lib/drizzle/io.ex

```diff
@@ -48,22 +53,26 @@ defmodule Drizzle.IO do
     {:noreply,
      state
      |> Enum.map(fn {zone_name, %{gpio: gpio, currstate: _cst}} ->
+
        {zone_name,
         %{
           gpio: gpio,
           currstate:
             cond do
```

Comment:
try to steer clear from so much inline code, turn your code into small functions that have a meaningful name when you find yourself inserting conditions like these :)

Reply:
ok, I've refactored it a bit, added a private function to handle the state change, looks way better now.